hpOneView/resources/fc_sans/san_managers.py

```diff
@@ -0,0 +1,142 @@
+__version__ = '0.0.1'
+__copyright__ = '(C) Copyright (2012-2016) Hewlett Packard Enterprise Development LP'
+__license__ = 'MIT'
+__status__ = 'Development'
+
+from hpOneView.resources.resource import ResourceClient
+
+
+class SanManagers(object):
+    URI = '/rest/fc-sans/device-managers'
+    PROVIDER_URI = '/rest/fc-sans/providers'
+
+    def __init__(self, con):
+        self._connection = con
+        self._client = ResourceClient(con, self.URI)
+
+    def get_all(self, start=0, count=-1, query='', sort=''):
+        """
+        Retrieves the list of registered SAN Managers
+
+        Args:
+            start:
+                The first item to return, using 0-based indexing.
+                If not specified, the default is 0 - start with the first available item.
+            count:
+                The number of resources to return. A count of -1 requests all the items. The actual number of items in
+                the response may differ from the requested count if the sum of start and count exceed the total number
+                of items, or if returning the requested number of items would take too long
+            query:
+                A general query string to narrow the list of resources returned.
+                The default is no query - all resources are returned.
+            sort:
+                The sort order of the returned data set. By default, the sort order is based
+                on create time, with the oldest entry first.
+
+        Returns:
+            list: A list of SAN managers
+
+        """
+        return self._client.get_all(start=start, count=count, query=query, sort=sort)
+
+    def get(self, id_or_uri):
+        """
+        Retrieves a single registered SAN Manager by id or uri
+
+        Args:
+            id_or_uri: Could be either the SAN Manager resource id or uri.
+
+        Returns:
+            dict: The SAN Manager resource.
+        """
+        return self._client.get(id_or_uri=id_or_uri)
+
+    def update(self, resource, id_or_uri):
+        """
+        Updates a registered Device Manager
+
+        Args:
+            id_or_uri: Could be either the Device manager id or uri.
+            resource (dict): Object to update.
+
+        Returns:
+            dict: The device manager resource.
+        """
+        return self._client.update(resource=resource, uri=id_or_uri)
+
+    def add(self, resource, provider_uri_or_id, timeout=-1):
+        """
+        Adds a Device Manager under the specified provider
+
+        Args:
+            resource: dict object to add
+            provider_uri_or_id: id or uri of provider
+            timeout:
+                Timeout in seconds. Wait task completion by default. The timeout does not abort the operation
+                in OneView, just stop waiting for its completion.
+
+        Returns:
+            dict: Added SAN Manager
+        """
+        add_client = ResourceClient(self._connection, self.PROVIDER_URI)
```

Comment:
Maybe should be better if you put this  on the constructor method, to not repeat the same line on the next two methods.